File: src/main/java/org/quang1807/citizenstrade/config/ConfigManager.java
package org.quang1807.citizenstrade.config;

import org.quang1807.citizenstrade.CitizenShop;
import org.bukkit.ChatColor;
import org.bukkit.configuration.file.FileConfiguration;
import org.bukkit.configuration.file.YamlConfiguration;

import java.io.File;
import java.io.IOException;

public class ConfigManager {
    private final CitizenShop plugin;
    private File configFile;
    private File guiConfigFile;
    private FileConfiguration config;
    private FileConfiguration guiConfig;

    public ConfigManager(CitizenShop plugin) {
        this.plugin = plugin;
        setupFiles();
    }

    private void setupFiles() {
        if (!plugin.getDataFolder().exists()) {
            plugin.getDataFolder().mkdirs();
        }

        configFile = new File(plugin.getDataFolder(), "config.yml");
        guiConfigFile = new File(plugin.getDataFolder(), "gui.yml");

        // Save default configurations if they don't exist
        if (!configFile.exists()) {
            plugin.saveResource("config.yml", false);
        }

        if (!guiConfigFile.exists()) {
            plugin.saveResource("gui.yml", false);
        }
    }

    public void loadConfigs() {
        try {
            config = YamlConfiguration.loadConfiguration(configFile);
            guiConfig = YamlConfiguration.loadConfiguration(guiConfigFile);
            plugin.getLogger().info("Configuration files loaded successfully!");
        } catch (Exception e) {
            plugin.getLogger().severe("Error loading configuration files: " + e.getMessage());
        }
    }

    public void saveConfigs() {
        try {
            if (config != null) {
                config.save(configFile);
            }
            if (guiConfig != null) {
                guiConfig.save(guiConfigFile);
            }
        } catch (IOException e) {
            plugin.getLogger().severe("Error saving configuration files: " + e.getMessage());
        }
    }

    public String getMessage(String key) {
        String message = config.getString("messages." + key, "&cMessage not found: " + key);
        return translateColors(message);
    }

    public String translateColors(String text) {
        return ChatColor.translateAlternateColorCodes('&', text);
    }

    // Getters
    public FileConfiguration getConfig() {
        return config;
    }

    public FileConfiguration getGuiConfig() {
        return guiConfig;
    }

    public File getConfigFile() {
        return configFile;
    }

    public File getGuiConfigFile() {
        return guiConfigFile;
    }
}

File: src/main/kotlin/org/quang1807/fancyshop/config/ConfigManager.kt
package org.quang1807.fancyshop.config

import org.quang1807.fancyshop.FancyShop
import org.bukkit.ChatColor
import org.bukkit.configuration.file.FileConfiguration
import org.bukkit.configuration.file.YamlConfiguration
import java.io.File
import java.io.IOException

class ConfigManager(private val plugin: FancyShop) {

    private lateinit var configFile: File
    private lateinit var guiConfigFile: File
    private lateinit var tradesFile: File

    lateinit var config: FileConfiguration
        private set
    lateinit var guiConfig: FileConfiguration
        private set
    lateinit var tradesConfig: FileConfiguration
        private set

    fun loadConfigs() {
        setupFiles()
        config = YamlConfiguration.loadConfiguration(configFile)
        guiConfig = YamlConfiguration.loadConfiguration(guiConfigFile)
        tradesConfig = YamlConfiguration.loadConfiguration(tradesFile)
    }

    private fun setupFiles() {
        if (!plugin.dataFolder.exists()) {
            plugin.dataFolder.mkdirs()
        }

        configFile = File(plugin.dataFolder, "config.yml")
        guiConfigFile = File(plugin.dataFolder, "gui.yml")
        tradesFile = File(plugin.dataFolder, "trades.yml")

        if (!configFile.exists()) {
            plugin.saveResource("config.yml", false)
        }

        if (!guiConfigFile.exists()) {
            plugin.saveResource("gui.yml", false)
        }

        if (!tradesFile.exists()) {
            try {
                tradesFile.createNewFile()
            } catch (e: IOException) {
                plugin.logger.severe("Could not create trades.yml: ${e.message}")
            }
        }
    }

    fun getMessage(key: String): String {
        return translateColors(config.getString("messages.$key", "&cMessage not found: $key") ?: "&cMessage not found: $key")
    }

    fun translateColors(text: String): String {
        return ChatColor.translateAlternateColorCodes('&', text)
    }

    fun saveTradesConfig() {
        try {
            tradesConfig.save(tradesFile)
        } catch (e: IOException) {
            plugin.logger.severe("Could not save trades.yml: ${e.message}")
        }
    }
}